src/util/http_utils.js

```diff
@@ -681,6 +682,20 @@ function authorize_session_token(req, options) {
     }
 }
 
+function authenticate_request_by_service(req, sdk_type) {
```

Comment:
perhaps just pass in the sdk object itself instead of its type?

Reply:
I think it's better to move this function to signature_utils or endpoint_utils

Reply:
1. I would pass it, although I think it is less readable than the enum.
2. I moved it to `signature_utils` ( I didn't choose `endpoint_utils` is not under `src/util` but `src/endpoint` and its current functions are only used by `endpoint.js`).